dev/readthedocs.md

```diff
@@ -4,28 +4,45 @@ Please visit the [docs](https://semesterly-v2.readthedocs.io/en/latest/index.htm
 answer the following questions.
 
 1. What is the command I run to get the courses from Fall 2021?
+Access the semesterly Docker container shell. Then use the command:
+`python manage.py ingest jhu --term Fall --years 2021`
+This command gets the course data from JHU, validates it, and writes it to formatted JSON
 
 2. How do I then load those courses into my database?
+`python manage.py digest jhu`
 
 3. How do I get a terminal running in my docker container?
+Click on the VSCode Docker extension, right click on the desired Docker container and click "Attach Shell".
 
 4. Where do I store data I don’t want to commit?
+You can stash data on a stack using the command `git stash`.
 
 5. What is our stack?
+React/Redux  (Frontend) + Django (Backend) + PostgreSQL (Database)
 
 6. What branch do I create a new branch off of when developing?
+`develop`
 
 7. If I want to start on a feature called myfeature, what should the branch name be?
+`feature/myfeature`
 
    What about if I want to refactor myreducer?
+   `refactor/myreducer`
 
 8. What is the preferred format for commit messages?
+"Topic: Message"
 
 9. What linters do we run against our code?
+eslint + prettier, black.
 
 10. What is the max line length set to?
+88, as per black defaults.
+
 
 11. What is a FeatureFlowView?
+A FeatureFlowView is a class that helps the frontend initialize when the user makes a request for a page.
+It can do this by creating data in the FeatureFlowView and passing the response as JSON.
+The frontend then uses the name of the flow and other data from the response to dispatch actions for the flow.
```

Comment:
Correct, the main purpose here is to draw attention to how the backend and frontend are formally connected. This is the view that renders the timetable you see at `jhu.semester.ly`.

Reply:
Ok, thanks. Just to make sure I'm getting it, this is what happens when a user goes to `jhu.semester.ly` and hits enter:

1. User makes a request for the FeatureFlowView, as specified by `urls.py`
2. FeatureFlowView gets a bunch of data that the frontend needs to load, and renders `timetable.html` with a JSON object with `init_data` gathered from the view for the client.

Can you briefly describe how/where the React components come in, or point me to some resources on Django+React? I think they get rendered into the `page` of `timetable.html` but not exactly sure how that works.

Reply:
When the server is run, all of the JS files (and thus the React app) are compiled by webpack, Through some kind of black magic (that I don't understand), the `initData` gets passed into `init.jsx` as a global variable, which then renders the `SemesterlyContainer` component, which is basically the root of the React app. 

```js
render(
  <Provider store={store}>
    <DndProvider backend={HTML5Backend}>
      <SemesterlyContainer />
    </DndProvider>
  </Provider>,
  document.getElementsByClassName("page")[0]
);
```
I think this is what renders it in the `page` div in `timetable.html`, but tbh I don't know how webpack works either. I haven't had to know how exactly it works to work on the app, which is why I guess I haven't figured it out yet.

Reply:
[This blog ](https://dev.to/zachtylr21/how-to-serve-a-react-single-page-app-with-django-1a1l) gave me some idea as to how `webpack` works for React + Django with`django-webpack-loader`. 

`initData` seems to get defined in `timetable.html`, but yeah I can't figure out how it works either...

Reply:
I'll take a look at the article tomorrow

Reply:
This is my understanding:
1. User makes a request for `jhu.semester.ly/`, and this endpoint is handled by the FeatureFlowView, as specified by urls.py.
2. FeatureFlowView gets a bunch of data that the frontend needs to load, and renders `timetable.html` with a JSON object with `init_data` gathered from the view for the client.
3. This `init_data` is used in `timetable.html` as a global variable `var initData = {{init_data|escapejs}}`. The `|escapejs` is a template filter, but I think all it does is make the `init_data` usable in JS.
4. Webpack compiles all of the JS files together and the template renders the bundle with `{% render_bundle 'main' 'js' %}`, a block statement from the django webpack loader library that we make use of
5. `init.jsx` is one of these files, and I suspect that since the `render` function isn't within a function, it is run directly. This means that the rendered bundle will now attach the `SemesterlyContainer` to the `page` div.
6. `SemesterlyContainer` is basically the app from there; `<Semesterly />` has the React components and the Container passes in redux state and dispatch functions.